contributors/devel/sig-instrumentation/logging.md

```diff
@@ -1,37 +1,73 @@
-## Logging Conventions
+## Logging
 
-The following conventions for the klog levels to use.
-[klog](http://godoc.org/github.com/kubernetes/klog) is globally preferred to
-[log](http://golang.org/pkg/log/) for better runtime control.
+This document will provide an overview on how Kubernetes community recommendation on logging. 
+Following recommendation should be followed in https://github.com/kubernetes/kubernetes, however any Kubernetes or external project is fill to follow guidelines below as most of the ideas can be applied more broadly.
 
-Shared libraries, such as `client-go`, should not use `klog.ErrorS()` and `klog.InfoS()`,
+### Logging in Kubernetes
+
+Kubernetes project uses [klog](https://github.com/kubernetes/klog) for logging.
+This library was created as a permanent fork of glog, as the original project was no longer developed. 
+However, should not be used by external projects as it is mostly developed to maintain backward compatibility.
+With introduction of [Structured Logging](https://github.com/kubernetes/enhancements/issues/1602) Kubernetes is being migrated to [logr](https://github.com/go-logr/logr) interface, any changes in klog are done to ensure smooth transition.
+
+For projects that want to integrate with Kubernetes or write same log format we recommend using [klogr](https://github.com/kubernetes/klog/tree/main/klogr), reimplementation of klog format that exposes [logr](https://github.com/go-logr/logr) interface.
+
+### Logging Conventions
+
+### When not to log?
+
+Shared libraries, such as `client-go`, should not log write logs themselves,
 but just return `error`, because client libraries may be used in CLI UIs that wish to control output.
 
-Please see the [Structured Logging Guide](migration-to-structured-logging.md#structured-logging-in-kubernetes) for more information on how to set keys and values in structured logs.
+### How to log
 
-* klog.ErrorS() - Errors should be used to indicate unexpected behaviours in code, like unexpected errors returned by subroutine function calls.
-Logs generated by `ErrorS` command may be enhanced with additional debug information (by logging library). Calling `ErrorS` with `nil` as error may be acceptable if there is error condition that deserves a stack trace at this origin point.
+There are two main klog methods for writing logs `klog.InfoS` and `klog.ErrorS` as they provide [logr](https://github.com/go-logr/logr) compatible interface.
+For those two functions exists additional flavors, like `klog.V(2).InfoS` that allows caller to increase log verbosity (making it only available when `--v=2` flag is provided to binary) and `klog.InfoSDepth` with `klog.ErorSDepth` that give control to change caller information.
```

Comment:
Is this accurate?
```suggestion
You can also find variants of those functions, such as `klog.V(2).InfoS` that allows caller to increase log verbosity (making it only available when `--v=2` flag is provided toa  binary). Another example is `klog.InfoSDepth` and `klog.ErorSDepth` that provide a way to report caller information.
```

Reply:
Part about caller information is wrong. Caller information is always provided, it points to code location that called log library. `klog.InfoSDepth` is meant to give some control over caller information to author of code.

Calling just `Info` works in cases where logging library is called directly, however in some cases it's useful to create logging utility functions that log themselves removing repetition or ensuring consistency. In such cases log library is called indirectly, resulting in caller information pointing to utility function. This becomes a problem when utility function is used frequently resulting caller information becoming useless as all different calls point to same place.